gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -182,26 +181,30 @@ gum_qnx_enumerate_ranges (pid_t pid,
   g_assert (fd != -1);
   g_free (as_path);
 
-  res = devctl (fd, DCMD_PROC_PAGEDATA, &membufs, sizeof (membufs),
-      &num_membufs);
+  res = devctl (fd, DCMD_PROC_PAGEDATA, 0, 0, &num_mapinfos);
   g_assert (res != 0);
 
-  close (fd);
+  mapinfos = malloc (sizeof (procfs_mapinfo) * num_mapinfos);
 
-  g_assert_cmpint (num_membufs, >, 512);
+  res = devctl (fd, DCMD_PROC_PAGEDATA, &mapinfos,
+                sizeof (procfs_mapinfo) * num_mapinfos, NULL);
+  g_assert (res != 0);
 
-  while (carry_on && i < num_membufs)
+  while (carry_on && i < num_mapinfos)
```

Comment:
We always use this style `i != num_mapinfos` elsewhere in Gum (the reasoning is that infinite loops are easier to detect than off-by-one errors).